File: tests/fixtures/generic/vulnerable/taint_ldap.js
// @audit-fixture
// @rule: taint_ldap
// @expected: detected
const ldap = require("ldapjs");
const express = require("express");
const app = express();
app.post("/auth", (req, res) => {
    const user = req.body.user;
    const client = ldap.createClient({ url: "ldap://localhost" });
    client.search("dc=example,dc=com", { filter: "(uid=" + user + ")" }, (err, r) => {
        res.json({ ok: !err });
    });
});

File: tests/fixtures/generic/vulnerable/taint_ldap.py
# @audit-fixture
# @rule: taint_ldap
# @category: security
# @expected: detected
from flask import request
import ldap
@app.route("/search")
def search():
    user = request.args.get("user")
    conn.search_s("dc=example", ldap.SCOPE_SUBTREE, f"(uid={user})")
